src/property.unit.ts

```diff
@@ -435,4 +458,306 @@ describe("Property", () => {
 			});
 		});
 	});
-});
\ No newline at end of file
+
+	describe("pendingInit", () => {
+		describe("value property", () => {
+			let valuePropModel: Model;
+
+			beforeAll(() => {
+				valuePropModel = new Model({
+					Person: {
+						Id: {
+							type: String,
+							identifier: true
+						},
+						Name: {
+							type: String
+						}
+					}
+				});
+			});
+
+			it("is true for a new object without a provided value", async () => {
+				const Person = valuePropModel.types.Person;
+				const instance = await Person.create({}) as any;
+				const property = Person.getProperty("Name");
+				expect(instance.Name).toBeNull();
+				expect(Property$pendingInit(instance, property)).toBe(true);
+			});
+
+			it("is false for a new object with a provided value", async () => {
+				const Person = valuePropModel.types.Person;
+				const instance = await Person.create({ Name: "Test" }) as any;
+				const property = Person.getProperty("Name");
+				expect(instance.Name).toBe("Test");
+				expect(Property$pendingInit(instance, property)).toBe(false);
+			});
+
+			it("is false for an existing object with a value", async () => {
+				const Person = valuePropModel.types.Person;
+				const instance = await Person.create({ Id: "3", Name: "Test" }) as any;
+				const property = Person.getProperty("Name");
+				expect(instance.Name).toBe("Test");
+				expect(Property$pendingInit(instance, property)).toBe(false);
+			});
+
+			it("is true for an existing object without a value", async () => {
+				const Person = valuePropModel.types.Person;
+				const instance = await Person.create({ Id: "4" }) as any;
+				const property = Person.getProperty("Name");
+				expect(instance.Name).toBeNull();
+				expect(Property$pendingInit(instance, property)).toBe(true);
+			});
```

Comment:
_Test suggestion_ an undefined/null string or a Boolean that's starts as false

Reply:
The same test but with a boolean value? I'm guessing you're suggesting this because this ensures that the pending init logic isn't affected by data types?

Reply:
This actually uncovered another edge case. When initial state is passed in for a new object the setter is called but if the value isn't changing it doesn't call `setValue`, so pending init wasn't getting set to false. I don't think this impacted product functionality though.